src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs

```diff
@@ -273,43 +273,60 @@ private static bool IsBoostedQuery(Query query)
 
             return false;
         }
-
-        private static Sort GetSort(SortedField[] sortedFields)
+        
+        private Sort GetSort(IndexQueryServerSide query)
         {
-            if (sortedFields == null || sortedFields.Length == 0)
+            var orderByFields = query.Fields.OrderBy;
+
+            if (orderByFields == null)
                 return null;
 
-            return new Sort(sortedFields.Select(x =>
+            var sort = new List<SortField>();
+
+            foreach (var field in orderByFields)
             {
                 var sortOptions = SortOptions.String;
 
-                if (x.Field == Constants.Documents.Indexing.Fields.IndexFieldScoreName)
-                    return SortField.FIELD_SCORE;
+                if (field.Name == Constants.Documents.Indexing.Fields.IndexFieldScoreName)
+                {
+                    sort.Add(SortField.FIELD_SCORE);
+                    continue;
+                }
 
-                if (InvariantCompare.IsPrefix(x.Field, Constants.Documents.Indexing.Fields.AlphaNumericFieldName, CompareOptions.None))
+                if (InvariantCompare.IsPrefix(field.Name, Constants.Documents.Indexing.Fields.AlphaNumericFieldName, CompareOptions.None))
                 {
-                    var customFieldName = SortFieldHelper.ExtractName(x.Field);
+                    var customFieldName = SortFieldHelper.ExtractName(field.Name);
                     if (customFieldName.IsNullOrWhiteSpace())
                         throw new InvalidOperationException("Alphanumeric sort: cannot figure out what field to sort on!");
 
                     var anSort = new AlphaNumericComparatorSource();
-                    return new SortField(customFieldName, anSort, x.Descending);
+                    sort.Add(new SortField(customFieldName, anSort, field.Ascending == false));
+                    continue;
                 }
 
-                if (InvariantCompare.IsPrefix(x.Field, Constants.Documents.Indexing.Fields.RandomFieldName, CompareOptions.None))
+                if (InvariantCompare.IsPrefix(field.Name, Constants.Documents.Indexing.Fields.RandomFieldName, CompareOptions.None))
                 {
-                    var customFieldName = SortFieldHelper.ExtractName(x.Field);
+                    var customFieldName = SortFieldHelper.ExtractName(field.Name);
                     if (customFieldName.IsNullOrWhiteSpace()) // truly random
-                        return new RandomSortField(Guid.NewGuid().ToString());
+                        sort.Add(new RandomSortField(Guid.NewGuid().ToString()));
```

Comment:
Same here: `ORDER BY random()` or `ORDER BY random(1234)` with the `1234` as the seed